earthaccess/store.py

```diff
@@ -188,7 +188,7 @@ def set_requests_session(
                 resp.raise_for_status()
             else:
                 self._requests_cookies.update(new_session.cookies.get_dict())
-        elif resp.status_code >= 200 and resp.status_code <= 300:
+        elif 200 <= resp.status_code <= 300:
```

Comment:
I think we want this instead:

```suggestion
        elif 200 <= resp.status_code < 300:
```

We also seem to be missing handling the rest of the 3xx and 4xx range.

Reply:
Good observation @chuckwondo! 
For the 3xx and 4xx handling, I would appreciate some suggestions, since I'm not sure myself :)

Reply:
my suggestion would be to remove the condition on the final `else` (which currently checks for code >= 500), so that everything else raises